### ISSUE TITLE
Installation "warning" messages on Anaconda 5.2

### ISSUE THREAD

**Issue by castillohair** [taborlab/FlowCal#298]
On Windows 10 (April 2018 Update) with Anaconda 5.2, running the installation script results in the following red "warning" (it's red and obnoxious but doesn't crash or stop anything) in the middle of all the text:

```
distributed 1.21.8 requires msgpack, which is not installed.
```

On OSX (High Sierra) with Anaconda 5.2, running the installation script produces these two red warnings:

```
distributed 1.21.8 requires msgpack, which is not installed.
grin 1.2.1 requires argparse>=1.1, which is not installed.
```

In both cases, installation proceeds normally despite these messages. The FlowCal examples run perfectly, and unit tests pass.

As a reminder, both installation scripts contain the following two lines:

```shell
python -m pip install -r requirements.txt --no-cache-dir
python setup.py install
```

I traced back the origin of the error to `pip` itself, the utility that we use to install dependencies. In both cases, running `python -m pip install -r requirements.txt --no-cache-dir` on a terminal produces the warnings.

`pip` as included in Anaconda 5.2 is in version 10.0.1 (see [here](https://docs.anaconda.com/anaconda/packages/py2.7_win-64) and [here](https://docs.anaconda.com/anaconda/packages/py3.6_win-64)). However the latest version is 18.0 (see [here](https://pypi.org/project/pip/)). Upgrading `pip` with the following command

```
python -m pip install --upgrade pip
```

Results in complete elimination of the warnings in both OSs.

The conclusions are:
- These warnings are coming from `pip`, and are not a problem with FlowCal. For this reason, I think the release of FlowCal can proceed without solving this.
- However, one easy thing that can be done to alleviate this is to upgrade pip in both installation scripts before installing requirements and FlowCal itself, i.e., the first line should be `python -m pip install --upgrade pip`.
- Lastly, `pip`'s huge version mismatch between the one included with Anaconda and the most recent one reveals that keeping `pip` up to date is not a priority for the Anaconda developers. Furthermore, `conda` is the package manager endorse by Anaconda and not `pip`. While the [official documentation](https://conda.io/docs/user-guide/tasks/manage-pkgs.html#installing-non-conda-packages) does not prevent using `pip` when `conda` is present, user reports [warn against this practice](https://stackoverflow.com/questions/20994716/what-is-the-difference-between-pip-and-conda). It seems that switching installation to `conda` might be the most correct thing to do and might alleviate potential issues in the long run. But I haven't read enough to be sure.

To summarize, the solution to this issue could be one of the following:
- Nothing, if we decide this is not a problem because it doesn't break FlowCal.
- Upgrade `pip` at the beginning of both installation scripts.
- Switch to `conda` for installing requirements.
- Something else.

**Comment by JS3xton** [comment]
I don't like upgrading `pip` at the beginning of installation scripts. That sounds like a hack that will only be necessary in the short term.

Using `conda` where possible seems like a good idea to me.

**Comment by castillohair** [comment]
The most recent version of Anaconda as of now (2018.12) includes `pip` 18.1, so this is not an issue anymore. Still, the idea of switching to `conda` might still be worth considering in the long run. I will close this issue and open a new one dedicated just to switching to `conda`.
